### PR TITLE
gh-44 Add support for task params

### PR DESCRIPTION
**Pull request by ghillert**
Resolves spring-cloud/spring-cloud-dataflow-ui#44

### CODE REVIEW

File: ui/app/scripts/task/services.js [hunk 1]
```diff
@@ -156,23 +156,34 @@ define(['angular'], function (angular) {
...
 
-            this.launch(taskLaunchRequest.taskName, parametersAsString);
+            this.launch(taskLaunchRequest.taskName, parametersAsString, argumentsAsString);
           },
-          launch: function (taskName, jsonDataAsString) {
+          launch: function (taskName, parametersAsString, argumentsAsString) {
             console.log('Launching task...' + taskName);
-            $resource($rootScope.dataflowServerUrl + '/tasks/deployments/:taskname', { 'taskname': taskName, 'properties': jsonDataAsString }, {
+            $resource($rootScope.dataflowServerUrl + '/tasks/deployments/:taskname', {
+              'taskname': taskName, 'properties': parametersAsString, 'params': argumentsAsString }, {
```
Comment (ilayaperumalg) [inline ui/app/scripts/task/services.js]: I think this is incorrect. The `parametersAsString` should be set for `params` and `argumentsAsString` for `properties`.

Comment (ghillert) [inline ui/app/scripts/task/services.js]: The naming is messed up but I think it is correct. Let me clarify the naming.

**Comment by ghillert** [comment]
Also @ilayaperumalg be aware of https://github.com/spring-cloud/spring-cloud-dataflow/issues/649

File: ui/app/scripts/task/views/launch.html [hunk 2]
```diff
@@ -1,46 +1,96 @@
...
 						<span class="glyphicon glyphicon-warning-sign form-control-feedback"
-								ng-show="taskParameterForm.key.$invalid && taskParameterForm.key.$touched"></span>
-						<p class="help-block" ng-show="taskParameterForm.key.$error.required && taskParameterForm.key.$touched">You must provide a key.</p>
+								ng-show="taskPropertyForm.key.$invalid && taskPropertyForm.key.$touched"></span>
+						<p class="help-block" ng-show="taskPropertyForm.key.$error.required && taskPropertyForm.key.$touched">You must provide a key.</p>
 					</div>
-					<div class="col-xs-10 col-sm-5" ng-class="taskParameterForm.value.$invalid && taskParameterForm.value.$touched ? 'has-warning has-feedback' : ''">
-						<input type="text" name="value" class="form-control" placeholder="Parameter Value" ng-model="taskParam.value" required>
+					<div class="col-xs-10 col-sm-5" ng-class="taskPropertyForm.value.$invalid && taskPropertyForm.value.$touched ? 'has-warning has-feedback' : ''">
+						<input type="text" name="value" class="form-control" placeholder="Parameter Value" ng-model="taskProperty.value" required>
```
Comment (ilayaperumalg) [inline ui/app/scripts/task/views/launch.html]: This should be `Property Value` in the placeholder.

File: ui/app/scripts/task/views/launch.html [hunk 3]
```diff
@@ -1,46 +1,96 @@
...
 			<div class="col-xs-2">
-				<button data-id="" type="button" class="btn btn-warning" data-dismiss="modal" ng-click="removeParameter(taskParam)"><span class="glyphicon glyphicon-remove"></span></button>
+				<button data-id="" type="button" class="btn btn-warning" data-dismiss="modal" ng-click="removeProperty(taskProperty)"><span class="glyphicon glyphicon-remove"></span></button>
 			</div>
 			<div class="col-xs-10">
 				<div class="form-group">
-					<div class="col-xs-10 col-sm-5" ng-class="taskParameterForm.key.$invalid && taskParameterForm.key.$touched ? 'has-warning has-feedback' : ''">
-						<input type="text" name="key" class="form-control" placeholder="Parameter Key" ng-model="taskParam.key" required>
+					<div class="col-xs-10 col-sm-5" ng-class="taskPropertyForm.key.$invalid && taskPropertyForm.key.$touched ? 'has-warning has-feedback' : ''">
+						<input type="text" name="key" class="form-control" placeholder="Parameter Key" ng-model="taskProperty.key" required>
```
Comment (ilayaperumalg) [inline ui/app/scripts/task/views/launch.html]: This needs to be `Property Key` at the placeholder.

Comment (ilayaperumalg) [inline ui/app/scripts/task/views/launch.html]: will fix both of them while merging.

**Comment by ilayaperumalg** [comment]
LGTM, squashed, rebased and merged as 8e98e220b9dcd906cc46d27d1b18dcd0738f7d81 along with 5e9e0b40cf899628e7ffe365898c452dd1c5fa3a

**Pull request closed without merging**